### ISSUE TITLE
TAS Studio Hang when Undo an Insert Frame

### ISSUE THREAD

**Issue by Meerkov** [TASVideos/BizHawk#1988]
### Summary
Emu Hawk hangs for a long time when you try to Undo inserting a frame into a long movie.

### Repro
[//]: # "If you can't figure out the list of steps, delete this section and put 'heisenbug' in the summary somewhere. If a Lua script can cause the bug, you can embed that instead (as simple as possible please)."
1. Get a movie thats at least 50k frames in TAS Studio
2. Insert a frame at 50,900 frames into the movie
3. Run for a few frames
4. Undo
5. Hang

### Result
Emu Hawk hangs for a long time (order of 30 minutes or more).
Check ram: See that Emuhawk is taking more than 3.2 GB of ram.

I think you get the same result if you Delete a frame, and undo as well.

### Host env.
[//]: # "List the computers you've found the bug with. If there's a version that doesn't have the bug, please put that in too. Here are some examples:"
- BizHawk 2.3.2; Win10

**Comment by vadosnaprimer** [comment]
And in latest release?

**Comment by YoshiRulz** [comment]
see also #1822 (closed because CNR)?

**Comment by adelikat** [comment]
CNR and I made a fix that probably fixes this issue, but since I couldn't reproduce to begin with it's somewhat a guess

**Comment by Meerkov** [comment]
On 2.4.1 I haven't gotten it to hang for 30 minutes yet. But I can get it to hang for sometimes 10-60 seconds.

**Comment by Meerkov** [comment]
Try inserting 20 frames one at a time. Then undo 20 times.

**Comment by adelikat** [comment]
Can you provide a .tasproj? and a frame for to do things and what things to do? I have not been able to reproduce

**Comment by RetroEdit** [comment]
I also have this issue, and had a `.tasproj` on hand to show it: [NEWFF1_2.4_HAND_RESYNC.zip](https://github.com/TASVideos/BizHawk/files/4610444/NEWFF1_2.4_HAND_RESYNC.zip)

It's an old Final Fantasy I & II Dawn of Souls TAS I was attempting to resync, though `.tasproj` itself is not old; it was imported from a `.bk2`.

I started the emulator paused. I inserted a single frame at 24181, then did ctrl-Z undo. BizHawk hung until I was forced to terminate the application (granted, my PC is not high-performance).
